gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/iterator/IteratorUtils.java

```diff
@@ -163,6 +192,7 @@ public static <K, S, V> Map<K, V> collectMap(final Iterator<S> iterator, final F
             final S obj = iterator.next();
             map.put(key.apply(obj), value.apply(obj));
         }
+        CloseableIterator.closeIterator(iterator);
```

Comment:
i guess the idea here is to just close the iterator as close as possible to where it was exhausted?

Reply:
All these methods reduce the nested iterator to a single result and then discard it. I am just making sure they close their delegate properly, otherwise we get resource leaks (especially if the underlying iterator is something that actually holds resources open, like a Traversal).